proto/tendermint/abci/types.proto

```diff
@@ -123,23 +123,32 @@ message RequestApplySnapshotChunk {
 }
 
 message RequestPrepareProposal {
-  bytes                   hash   = 1;
-  tendermint.types.Header header = 2 [(gogoproto.nullable) = false];
+  bytes                     hash   = 1;
+  int64                     height = 2;
+  google.protobuf.Timestamp time   = 3 [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
+
   // txs is an array of transactions that will be included in a block,
   // sent to the app for possible modifications.
-  repeated bytes     txs                  = 3;
-  ExtendedCommitInfo local_last_commit    = 4 [(gogoproto.nullable) = false];
-  repeated Evidence  byzantine_validators = 5 [(gogoproto.nullable) = false];
+  repeated bytes     txs                  = 4;
+  ExtendedCommitInfo local_last_commit    = 5 [(gogoproto.nullable) = false];
+  repeated Evidence  byzantine_validators = 6 [(gogoproto.nullable) = false];
+  // address of the public key of the node receiving this PrepareProposal request.
+  bytes proposer_address     = 7;
+  bytes next_validators_hash = 8;
   // the modified transactions cannot exceed this size.
-  int64 max_tx_bytes = 6;
+  int64 max_tx_bytes = 9;
 }
 
 message RequestProcessProposal {
-  bytes                   hash                 = 1;
-  tendermint.types.Header header               = 2 [(gogoproto.nullable) = false];
-  repeated bytes          txs                  = 3;
-  CommitInfo              proposed_last_commit = 4 [(gogoproto.nullable) = false];
-  repeated Evidence       byzantine_validators = 5 [(gogoproto.nullable) = false];
+  bytes                     hash                 = 1;
+  int64                     height               = 2;
+  google.protobuf.Timestamp time                 = 3 [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
```

Comment:
Very minor.
Would it make sense to also put these together with `proposer_address and` `next_validators_hash` at the end?
I'm suggesting to put them at the end, in case we need to add another later, so they all still appear together

Reply:
Done.